x-pack/test/functional/apps/lens/group2/table.ts

```diff
@@ -41,6 +41,32 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       });
     });
 
+    it('should apply density changes correctly', async () => {
+      await lens.openVisualOptions();
+      await lens.setDataTableDensity('compact');
+      expect(
+        await find.existsByCssSelector(
+          `[data-test-subj="lnsDataTable"][class*="cellPadding-s-fontSize-s"]`
+        )
+      ).to.be(true);
+
+      await lens.openVisualOptions();
+      await lens.setDataTableDensity('expanded');
+      expect(
+        await find.existsByCssSelector(
+          `[data-test-subj="lnsDataTable"][class*="cellPadding-l-fontSize-l"]`
+        )
```

Comment:
It may make sense to encapsulate it into a lens page method:

```ts
async checkDataTableDensity(size: 'l' | 'm' | 's'){
  await find.existsByCssSelector(
          `[data-test-subj="lnsDataTable"][class*="cellPadding-${size}-fontSize-${size}"]`
        );
}
```